### PR TITLE
DT-67 - improve reason field on batch operation modal

### PR DESCRIPTION
**Pull request by rossedfort**
## Description & motivation 💭 
This improves the UX in the batch operation confirmation modal, specifically: 
- make reason optional
- provide placeholder with user email (if available) and Web UI context
- use placeholder if no reason is entered, and append placeholder if custom reason is entered

This is so that users don't have to enter a reason even though the API requires it. It also has the benefit of adding some more meaningful context to batch operations.

### Screenshots (if applicable) 📸 
Screen Shot 2023-03-16 at 11 05 24 AM
Screen Shot 2023-03-16 at 11 05 31 AM
Screen Shot 2023-03-16 at 11 06 04 AM

### Design Considerations 🎨 
N/a
### How was this tested 👻 
- [x] Manual testing
- [x] E2E tests added
- [ ] Unit tests added

### Steps for others to test: 🚶🏽‍♂️🚶🏽‍♀️ 
- with the Web UI running against temporal cli version >= 0.6.0, open the recent workflows page, select some workflows, and click the "Request Cancellation" or "Terminate" buttons in the table header
- Ensure the correct placeholder information is in the "reason" Input.
- Ensure that the reason is properly forwarded to the API when left blank or when a custom reason is entered.
### Issue(s) closed 
DT-67
## Docs
N/a

### CODE REVIEW

File: src/lib/components/workflow/batch-operation-confirmation-modal.svelte [hunk 1]
```diff
@@ -17,25 +21,31 @@
     confirm: { reason: string };
   }>();
 
+  $: confirmText = action === 'Cancel' ? 'Confirm' : action;
+  $: placeholder = `${pastTense(action)} from Web UI${
+    $authUser?.email ? ` by ${$authUser.email}` : ''
```
Comment (Alex-Tideman) [inline src/lib/components/workflow/batch-operation-confirmation-modal.svelte:26]: What about the case when it's local development and there is no authUser?

**Comment by rossedfort** [comment]
> Do we want to append that message when there is a custom reason? Not sure if there's a hard requirement around that or not but I'd think some users would want to be able to set the message to exactly what they want.

Yeah great point, Matt noted in the JIRA ticket for this that we should allow users to customize this message, just a matter of whether we want to do it now or later. I'm game for doing it now, but it would add a bit of scope and this smaller change may not make it for next release.

File: src/lib/components/workflow/batch-operation-confirmation-modal.svelte (hunk 1 above)
Comment (rossedfort) [reply src/lib/components/workflow/batch-operation-confirmation-modal.svelte:26]: I think this should be handled with my ternary: `$authUser?.email ? ` by ${$authUser.email}` : ''`, If there is no email, the message will just be `(Terminated|Canceled) from Web UI`. Should we add an IP address or something?

Comment (Alex-Tideman) [reply src/lib/components/workflow/batch-operation-confirmation-modal.svelte:26]: durrr my bad

**Review by Alex-Tideman — APPROVED** [review]

**Pull request merged**